builder/googlecompute/config_test.go

```diff
@@ -685,17 +684,45 @@ func TestConfigExtraBlockDevice_create_image_multiple(t *testing.T) {
 	}
 }
 
+func TestConfigPrepareImageArchitecture(t *testing.T) {
+	cases := []struct {
+		Architecture string
+		Valid        bool
+	}{
+		{"x86_64", true},
+		{"ARM64", true},
+		{"i386", false}, // Assuming 'i386' is not supported, adjust as needed
+	}
+
+	for _, tc := range cases {
+		raw, tempfile := testConfig(t)
+		defer os.Remove(tempfile)
+
+		raw["image_architecture"] = tc.Architecture
+
+		var c Config
+		_, errs := c.Prepare(raw)
+
+		if tc.Valid && errs != nil {
+			t.Errorf("Expected architecture '%s' to be valid, but got error: %s", tc.Architecture, errs)
+		} else if !tc.Valid && errs == nil {
+			t.Errorf("Expected architecture '%s' to be invalid, but got no error", tc.Architecture)
+		}
+	}
+}
+
 // Helper stuff below
 
 func testConfig(t *testing.T) (config map[string]interface{}, tempAccountFile string) {
 	tempAccountFile = testAccountFile(t)
 
 	config = map[string]interface{}{
-		"credentials_file": tempAccountFile,
-		"project_id":       "hashicorp",
-		"source_image":     "foo",
-		"ssh_username":     "root",
-		"image_family":     "bar",
+		"credentials_file":   tempAccountFile,
+		"project_id":         "hashicorp",
+		"source_image":       "foo",
+		"ssh_username":       "root",
+		"image_architecture": "",
```

Comment:
I'd think you don't need to change that, string are always the empty string by default

Reply:
I removed the var as its not required.